lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -134,7 +141,14 @@ static public function convertPositionalToNamedPlaceholders($statement)
      */
     public function bindValue($param, $value, $type = null)
     {
-        return $this->bindParam($param, $value, $type, null);
+        if ($value instanceof OCI8Descriptor) {
```

Comment:
This is somewhat against the `Statement::bindValue()` contract which says that `$value` is the actual value to be bound. I see that `OCI8Descriptor` "represents" this particular value but this will break the usage of the DBAL OCI8 driver as standalone component. Furthermore in such case the `$type` parameter is not declaring anymore HOW the value is to be bound (because it is ignored). How should the user know how to properly use this API? It does not conform to the given contract anymore.
We need to find a solution that conforms to the contract and encapsulates the logic inside this method. I still wonder why we need to use descriptors for `CLOB` bindings at all. Can't we simply use the value + `\PDO::PARAM_STR`? Is there a failing test case? If that would be the case how does the driver determine whether a descriptor is necessary or not? Does it evaluate the given `$value` or is it bound to the actual table column type? I we cannot use `PARAM_STR` without descriptors at all anymore (which would be really weird), we should stick to always using `CLOB` descriptors in that case. `\PDO::PARAM_LOB` should then as already said always use a `BLOB` descriptor.